nilearn/datasets/func.py

```diff
@@ -1276,3 +1276,192 @@ def fetch_mixed_gambles(n_subjects=1, data_dir=None, url=None, resume=True,
         X, y, mask_img = _load_mixed_gambles(map(nibabel.load, data.zmaps))
         data.zmaps, data.gain, data.mask_img = X, y, mask_img
     return data
+
+
+def fetch_megatrawls_netmats(data_dir=None, choice_dimensionality=None,
+                             choice_timeseries=None,
+                             resume=True, verbose=1):
+    """Downloads and fetches Network Matrices data from MegaTrawls release in HCP.
+
+    This data can be used to predict relationships between imaging data (functional
+    connectivity) and non-imaging behavioural measures such as age, sex, education, etc.
+    The network matrices are estimated from functional connectivity datasets of 461
+    subjects [1]. Full technical details in [2].
+
+    The network matrices denoted as 'netmats' are estimated using full correlation
+    denoted as 'Znet1' or partial correlation with limited L2 regularisation
+    denoted as 'Znet2'. [1]
+
+    .. versionadded:: 0.1.5
+
+    Parameters
+    ----------
+    data_dir: string, default is None, optional
+        Path of the data directory. Used to force data storage in a specified
+        location.
+
+    choice_dimensionality: a string or list of strings or None, default is None
```

Comment:
`choice_dimensionality` -> why not `dimensionality` or even `dimension` ?
`choice_timeseries` -> `regression_mode`
I would also replace 'eigenregression' with 'eigen_regression' and 'multiple-spatial-regression' with 'multiple_spatial_regression'